# Importing libraries

In [1]:
import numpy as np
import pandas as pd

# 2. Data Quality Issues and data general insights

## Exploring data

I first start by looking at the data using .info() method, so that I have a general understanding of the dataset

In [2]:
pd.options.display.max_rows = 8
homicidios_2018 = pd.read_csv('./dataset/Homicidios_2018.csv', encoding = 'utf-8')
homicidios_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4966 entries, 0 to 4965
Data columns (total 20 columns):
Fecha                 4965 non-null object
Departamento          4965 non-null object
Municipio             4965 non-null object
Día                   4965 non-null object
Hora                  4965 non-null object
Barrio                4965 non-null object
Zona                  4965 non-null object
Clase de sitio        4965 non-null object
Arma empleada         4965 non-null object
Móvil Agresor         4965 non-null object
Móvil Victima         4965 non-null object
Edad                  4965 non-null float64
Sexo                  4965 non-null object
Estado civil          4965 non-null object
País de nacimiento    4965 non-null object
Clase de empleado     4965 non-null object
Profesión             4965 non-null object
Escolaridad           4965 non-null object
Código DANE           4965 non-null float64
Cantidad              4966 non-null int64
dtypes: float64(2), int64(1), ob

Then by using .isna().any(axis=1) or .isnull().any(axis=1) I find there is one row with no information, so that it needs to be deleted from the dataset using .dropna()

In [3]:
homicidios_2018['Municipio'].isna().value_counts()

False    4965
True        1
Name: Municipio, dtype: int64

In [4]:
homicidios_2018[homicidios_2018.isna().any(axis=1) | homicidios_2018.isnull().any(axis=1)]

Fecha Departamento Municipio  Día Hora Barrio Zona Clase de sitio  \
4965   NaN          NaN       NaN  NaN  NaN    NaN  NaN            NaN   

     Arma empleada Móvil Agresor Móvil Victima  Edad Sexo Estado civil  \
4965           NaN           NaN           NaN   NaN  NaN          NaN   

     País de nacimiento Clase de empleado Profesión Escolaridad  Código DANE  \
4965                NaN               NaN       NaN         NaN          NaN   

      Cantidad  
4965     12667

In [5]:
homicidios_2018 = homicidios_2018.dropna()

In [6]:
#Updated dataset
homicidios_2018.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4965 entries, 0 to 4964
Data columns (total 20 columns):
Fecha                 4965 non-null object
Departamento          4965 non-null object
Municipio             4965 non-null object
Día                   4965 non-null object
Hora                  4965 non-null object
Barrio                4965 non-null object
Zona                  4965 non-null object
Clase de sitio        4965 non-null object
Arma empleada         4965 non-null object
Móvil Agresor         4965 non-null object
Móvil Victima         4965 non-null object
Edad                  4965 non-null float64
Sexo                  4965 non-null object
Estado civil          4965 non-null object
País de nacimiento    4965 non-null object
Clase de empleado     4965 non-null object
Profesión             4965 non-null object
Escolaridad           4965 non-null object
Código DANE           4965 non-null float64
Cantidad              4965 non-null int64
dtypes: float64(2), int64(1), ob

The first procedure is to have an overview of the data, I will do so by using the value_counts() method which gives us information of the data in the dataset for a desired column and the frequency of that value

In [7]:
#Checking if there are any duplicated values
homicidios_2018.duplicated().value_counts()

False    4965
dtype: int64

In [8]:
# dates which are consistently repeated and its frequency
homicidios_2018['Fecha'].value_counts()

01/01/2018 12:00:00 AM    82
08/07/2018 12:00:00 AM    72
08/04/2018 12:00:00 AM    67
08/12/2018 12:00:00 AM    67
                          ..
06/09/2018 12:00:00 AM    20
03/10/2018 12:00:00 AM    19
09/10/2018 12:00:00 AM    19
05/01/2018 12:00:00 AM    18
Name: Fecha, Length: 144, dtype: int64

It can be seen that January 1st (Public holiday) is the day with the highest amount of homicides in Colombia in 2018, followed by July 8th (Sunday) and December 8th (Public holiday)
- 01/01/2018 12:00:00 AM    82
- 08/07/2018 12:00:00 AM    72
- 08/12/2018 12:00:00 AM    67

There is no data quality issues for this column, the data look consistent.

In [9]:
top3_dep = homicidios_2018['Departamento'].value_counts()
top3_dep

ANTIOQUIA       926
VALLE           889
CUNDINAMARCA    581
CAUCA           282
               ... 
GUAVIARE          5
AMAZONAS          4
VAUPÉS            2
GUAINÍA           1
Name: Departamento, Length: 32, dtype: int64

In [10]:
total_top3_dep = top3_dep[0]+top3_dep[1]+top3_dep[2]
percentage_dep = total_top3_dep/4965
percentage_dep

0.48257804632426987

The departments with the highest homicide number are Antioquia (925), Valle (889) and Cundinamarca (581), respectively. These departments represent take 48% of the homicides in Colombia in 2018. There is no any data issues here.

In [11]:
homicidios_2018['Municipio'].value_counts()

CALI (CT)            489
BOGOTÁ D.C. (CT)     431
MEDELLÍN (CT)        201
BARRANQUILLA (CT)    120
                    ... 
CURITÍ                 1
MONTEBELLO             1
PAJARITO               1
SAN FRANCISCO          1
Name: Municipio, Length: 582, dtype: int64

Cali, Bogota and Medellin are the cities with the highest number of homicides during 2018.

In [12]:
homicidios_2018['Día'].value_counts()

Domingo      1087
Sábado        812
Lunes         771
Jueves        610
Miércoles     581
Viernes       568
Martes        536
Name: Día, dtype: int64

A great number of homicides are committed on Sunday or Saturday. It is important to notice how monday is at the top 3 and friday is the second least day with homicides.

In [13]:
homicidios_2018['Hora'].value_counts().head()

20:30:00    80
20:00:00    79
21:30:00    71
23:00:00    65
19:30:00    64
Name: Hora, dtype: int64

In [14]:
homicidios_2018['Hora'].value_counts().tail()

3:41:00     1
10:52:00    1
12:12:00    1
16:29:00    1
21:49:00    1
Name: Hora, dtype: int64

The data looks consistent, although we can create subgroups to classify them and manage it in a different way.

In [15]:
homicidios_2018['Barrio'].value_counts()

CENTRO                       108
VEREDA                        32
EL BOSQUE                     19
LA ESPERANZA                  19
                            ... 
BALCON DEL LLANO               1
VEREDA CAÑO LA SAL             1
BARRIO SUEÑOS DE LIBERTAD      1
LOS PARQUES                    1
Name: Barrio, Length: 2947, dtype: int64

'Centro' or the city center remains as the sector of the city where most homicides are committed

In [16]:
homicidios_2018['Zona'].value_counts()

URBANA    3321
RURAL     1644
Name: Zona, dtype: int64

This data seems pretty good to pass it through to the next stage.

In [17]:
homicidios_2018['Clase de sitio'].value_counts()

VIAS PUBLICAS                    2990
FINCAS Y SIMILARES                561
CASAS DE HABITACION               300
CARRETERA                         144
                                 ... 
OFICINAS                            1
ESTACION TRANSMILENIO MOLINOS       1
TRANSPORTE INTERMUNICIPAL           1
COMPRAVENTA                         1
Name: Clase de sitio, Length: 111, dtype: int64

In [18]:
homicidios_2018['Arma empleada'].value_counts()

ARMA DE FUEGO                  3549
ARMA BLANCA / CORTOPUNZANTE    1133
CONTUNDENTES                    217
CUERDA/SOGA/CADENA               14
                               ... 
NO REPORTADO                      3
SIN EMPLEO DE ARMAS               1
MOTO BOMBA                        1
QUIMICOS                          1
Name: Arma empleada, Length: 15, dtype: int64

In [19]:
homicidios_2018['Móvil Agresor'].value_counts()

A PIE                    4129
PASAJERO MOTOCICLETA      620
CONDUCTOR MOTOCICLETA     107
PASAJERO VEHICULO          29
                         ... 
-                          10
CONDUCTOR TAXI              7
PASAJERO BARCO              7
PASAJERO BUS                3
Name: Móvil Agresor, Length: 11, dtype: int64

In [20]:
homicidios_2018['Móvil Victima'].value_counts()

A PIE                    4553
CONDUCTOR MOTOCICLETA     185
CONDUCTOR VEHICULO         95
PASAJERO MOTOCICLETA       33
                         ... 
PASAJERO TAXI               9
PASAJERO BARCO              8
PASAJERO BUS                4
CONDUCTOR BUS               2
Name: Móvil Victima, Length: 12, dtype: int64

The majority of homicides are commited by walk and the affected person also is nurdered while walking. There is no given data for these 2 as seen with the element with '-' in the tables above.

In [21]:
homicidios_2018['Edad'].describe()

count    4965.000000
mean       32.758711
std        13.108539
min         1.000000
25%        23.000000
50%        30.000000
75%        39.000000
max       100.000000
Name: Edad, dtype: float64

By analysing some statistics from Age, I found the minimum age was 1, which might be logical, however when displaying all details for this specific aspect, I found out that 'sort of Employee' for one of them was independent, then it clearly seems no logical.

In [22]:
homicidios_2018[homicidios_2018['Edad'] == 1]

Fecha Departamento       Municipio     Día     Hora  \
704   10/02/2018 12:00:00 AM      CAQUETÁ  FLORENCIA (CT)  Sábado  1:20:00   
2871  12/07/2018 12:00:00 AM    ANTIOQUIA        APARTADÓ  Jueves  0:00:00   

        Barrio    Zona                      Clase de sitio Arma empleada  \
704   RICAURTE  URBANA  HOTELES, RESIDENCIAS, Y SIMILARES.  CONTUNDENTES   
2871    OBRERO  URBANA                 CASAS DE HABITACION  CONTUNDENTES   

     Móvil Agresor Móvil Victima  Edad      Sexo Estado civil  \
704          A PIE         A PIE   1.0  FEMENINO      SOLTERO   
2871         A PIE         A PIE   1.0  FEMENINO      SOLTERO   

     País de nacimiento Clase de empleado Profesión   Escolaridad  \
704            COLOMBIA       DESEMPLEADO         -    ANALFABETA   
2871           COLOMBIA     INDEPENDIENTE         -  NO REPORTADO   

      Código DANE  Cantidad  
704    18001000.0         1  
2871    5045000.0         1

In [23]:
# Given that these details seem no ideal, then I condider deleting this row.
homicidios_2018 = homicidios_2018.drop(homicidios_2018.index[2871])
homicidios_2018.reset_index(drop=True)

Fecha Departamento            Municipio        Día  \
0     01/01/2018 12:00:00 AM     AMAZONAS         LETICIA (CT)      Lunes   
1     01/01/2018 12:00:00 AM     AMAZONAS         LETICIA (CT)      Lunes   
2     01/01/2018 12:00:00 AM    ANTIOQUIA                ANDES      Lunes   
3     01/01/2018 12:00:00 AM    ANTIOQUIA                ANDES      Lunes   
...                      ...          ...                  ...        ...   
4960  12/12/2018 12:00:00 AM    SANTANDER      BARRANCABERMEJA  Miércoles   
4961  12/12/2018 12:00:00 AM        VALLE            CALI (CT)  Miércoles   
4962  12/12/2018 12:00:00 AM        VALLE            CALI (CT)  Miércoles   
4963  12/12/2018 12:00:00 AM        VALLE  GUADALAJARA DE BUGA  Miércoles   

          Hora                   Barrio    Zona      Clase de sitio  \
0      9:45:00                ESPERANZA  URBANA       VIAS PUBLICAS   
1      9:45:00                ESPERANZA  URBANA       VIAS PUBLICAS   
2      8:30:00          ALTO DEL CEDRON   RURAL       VIAS PUBLICAS   
3      2:30:00         PARQUE PRINCIPAL   RURAL             PARQUES   
...        ...                      ...     ...                 ...   
4960  20:09:00    MIRADORES DEL CACIQUE  URBANA       VIAS PUBLICAS   
4961  19:47:00     REPUBLICA ISRAEL E16  URBANA       VIAS PUBLICAS   
4962  19:47:00     REPUBLICA ISRAEL E16  URBANA       VIAS PUBLICAS   
4963  17:45:00  CORR. CHAMBIMBAL/CAMPIÑ   RURAL  FINCAS Y SIMILARES   

                    Arma empleada         Móvil Agresor Móvil Victima  Edad  \
0     ARMA BLANCA / CORTOPUNZANTE                 A PIE         A PIE  16.0   
1     ARMA BLANCA / CORTOPUNZANTE                 A PIE         A PIE  17.0   
2                   ARMA DE FUEGO                 A PIE         A PIE  26.0   
3     ARMA BLANCA / CORTOPUNZANTE                 A PIE         A PIE  30.0   
...                           ...                   ...           ...   ...   
4960                ARMA DE FUEGO  PASAJERO MOTOCICLETA         A PIE  58.0   
4961                ARMA DE FUEGO                 A PIE         A PIE  32.0   
4962                ARMA DE FUEGO                 A PIE         A PIE  39.0   
4963  ARMA BLANCA / CORTOPUNZANTE                 A PIE         A PIE  16.0   

           Sexo Estado civil País de nacimiento    Clase de empleado  \
0     MASCULINO      SOLTERO             BRASIL  EMPLEADO PARTICULAR   
1     MASCULINO      SOLTERO             BRASIL  EMPLEADO PARTICULAR   
2     MASCULINO      SOLTERO           COLOMBIA        INDEPENDIENTE   
3     MASCULINO      SOLTERO           COLOMBIA           AGRICULTOR   
...         ...          ...                ...                  ...   
4960  MASCULINO      SOLTERO           COLOMBIA  EMPLEADO PARTICULAR   
4961   FEMENINO      SOLTERO           COLOMBIA          AMA DE CASA   
4962  MASCULINO      SOLTERO           COLOMBIA  EMPLEADO PARTICULAR   
4963  MASCULINO      SOLTERO           COLOMBIA          DESEMPLEADO   

     Profesión Escolaridad  Código DANE  Cantidad  
0            -  SECUNDARIA   91001000.0         1  
1            -  SECUNDARIA   91001000.0         1  
2            -    PRIMARIA    5034000.0         1  
3            -    PRIMARIA    5034000.0         1  
...        ...         ...          ...       ...  
4960         -    PRIMARIA   68081000.0         1  
4961         -  SECUNDARIA   76001000.0         1  
4962         -  SECUNDARIA   76001000.0         1  
4963         -  SECUNDARIA   76111000.0         1  

[4964 rows x 20 columns]

In [24]:
(homicidios_2018['Edad'] >= 90).value_counts()

False    4963
True        1
Name: Edad, dtype: int64

In [25]:
homicidios_2018[homicidios_2018['Edad']>=80]

Fecha Departamento            Municipio        Día  \
135   03/01/2018 12:00:00 AM    RISARALDA  SANTA ROSA DE CABAL  Miércoles   
648   08/02/2018 12:00:00 AM        VALLE            CALI (CT)     Jueves   
873   02/03/2018 12:00:00 AM       BOYACÁ             SOGAMOSO    Viernes   
987   05/03/2018 12:00:00 AM        CAUCA        PUERTO TEJADA      Lunes   
...                      ...          ...                  ...        ...   
4213  04/11/2018 12:00:00 AM       NARIÑO              IPIALES    Domingo   
4326  08/11/2018 12:00:00 AM        HUILA             PITALITO     Jueves   
4657  05/12/2018 12:00:00 AM    ANTIOQUIA             RIONEGRO  Miércoles   
4802  08/12/2018 12:00:00 AM    SANTANDER                GIRÓN     Sábado   

          Hora                    Barrio    Zona       Clase de sitio  \
135   20:50:00  CORREGIMIENTO LA CAPILLA   RURAL   FINCAS Y SIMILARES   
648   12:45:00    LOS CONQUISTADORES E11  URBANA         RESTAURANTES   
873   23:20:00                    CENTRO  URBANA  CASAS DE HABITACION   
987   10:30:00     CARLOS ALBERTO GUZMAN  URBANA        VIAS PUBLICAS   
...        ...                       ...     ...                  ...   
4213  18:00:00                 CHAMPAGTH  URBANA        VIAS PUBLICAS   
4326  10:00:00                 GUADUALES  URBANA  CASAS DE HABITACION   
4657  17:30:00                EL LLANITO  URBANA        VIAS PUBLICAS   
4802  18:00:00        HACIENDA LA MESETA  URBANA  CASAS DE HABITACION   

                    Arma empleada Móvil Agresor Móvil Victima  Edad  \
135                  CONTUNDENTES         A PIE         A PIE  83.0   
648                 ARMA DE FUEGO         A PIE         A PIE  83.0   
873                  CONTUNDENTES         A PIE         A PIE  80.0   
987                 ARMA DE FUEGO         A PIE         A PIE  83.0   
...                           ...           ...           ...   ...   
4213  ARMA BLANCA / CORTOPUNZANTE         A PIE         A PIE  80.0   
4326                 CONTUNDENTES         A PIE         A PIE  88.0   
4657                 CONTUNDENTES         A PIE         A PIE  86.0   
4802                 CONTUNDENTES         A PIE         A PIE  80.0   

           Sexo Estado civil País de nacimiento    Clase de empleado  \
135   MASCULINO     SEPARADO           COLOMBIA        INDEPENDIENTE   
648   MASCULINO        VIUDO           COLOMBIA        INDEPENDIENTE   
873   MASCULINO       CASADO           COLOMBIA           PENSIONADO   
987    FEMENINO        VIUDO           COLOMBIA        INDEPENDIENTE   
...         ...          ...                ...                  ...   
4213  MASCULINO      SOLTERO           COLOMBIA           PENSIONADO   
4326   FEMENINO      SOLTERO           COLOMBIA  EMPLEADO PARTICULAR   
4657  MASCULINO       CASADO           COLOMBIA  EMPLEADO PARTICULAR   
4802  MASCULINO        VIUDO           COLOMBIA           PENSIONADO   

         Profesión Escolaridad  Código DANE  Cantidad  
135              -    PRIMARIA   66682000.0         1  
648              -  SECUNDARIA   76001000.0         1  
873              -  SECUNDARIA   15759000.0         1  
987              -    PRIMARIA   19573000.0         1  
...            ...         ...          ...       ...  
4213             -  SECUNDARIA   52356000.0         1  
4326  NO REPORTADA    PRIMARIA   41551000.0         1  
4657             -  SECUNDARIA    5615000.0         1  
4802             -    PRIMARIA   68307000.0         1  

[21 rows x 20 columns]

In [26]:
homicidios_2018['Sexo'].value_counts()

MASCULINO    4555
FEMENINO      409
Name: Sexo, dtype: int64

In [27]:
homicidios_2018['Estado civil'].value_counts()

SOLTERO        3141
UNION LIBRE    1502
CASADO          241
SEPARADO         38
DIVORCIADO       17
-                13
VIUDO            12
Name: Estado civil, dtype: int64

In [28]:
homicidios_2018[homicidios_2018['Estado civil'] == '-']

Fecha        Departamento         Municipio        Día  \
870   02/03/2018 12:00:00 AM           ANTIOQUIA             TURBO    Viernes   
982   05/03/2018 12:00:00 AM           ANTIOQUIA             YONDÓ      Lunes   
2213  06/06/2018 12:00:00 AM           ANTIOQUIA       LA ESTRELLA  Miércoles   
2630  05/07/2018 12:00:00 AM  NORTE DE SANTANDER       CÚCUTA (CT)     Jueves   
...                      ...                 ...               ...        ...   
3447  03/09/2018 12:00:00 AM           SANTANDER  BUCARAMANGA (CT)      Lunes   
3819  04/10/2018 12:00:00 AM              ARAUCA              TAME     Jueves   
3852  05/10/2018 12:00:00 AM           ANTIOQUIA     MEDELLÍN (CT)    Viernes   
4785  08/12/2018 12:00:00 AM        CUNDINAMARCA            SOACHA     Sábado   

          Hora                  Barrio    Zona      Clase de sitio  \
870   19:40:00    VDA. BOCAS DE ATRATO   RURAL  FINCAS Y SIMILARES   
982   11:30:00        VDA. SANTA CLARA   RURAL       VIAS PUBLICAS   
2213  19:00:00         VDA. EL ROMERAL   RURAL       VIAS PUBLICAS   
2630   8:30:00  BARRIO LLERAS RESTREPO  URBANA        CANALIZACION   
...        ...                     ...     ...                 ...   
3447   1:41:00                 ALARCON  URBANA       VIAS PUBLICAS   
3819  16:30:00       VEREDA EL BOTALON   RURAL          CEMENTERIO   
3852  23:41:00      LA CANDELARIA C-10  URBANA       VIAS PUBLICAS   
4785  11:00:00         LAGOS DE MALIBU  URBANA       VIAS PUBLICAS   

                    Arma empleada         Móvil Agresor Móvil Victima  Edad  \
870                 ARMA DE FUEGO                 A PIE         A PIE  22.0   
982                 ARMA DE FUEGO                 A PIE         A PIE  55.0   
2213                ARMA DE FUEGO                 A PIE         A PIE  30.0   
2630  ARMA BLANCA / CORTOPUNZANTE                 A PIE         A PIE  28.0   
...                           ...                   ...           ...   ...   
3447  ARMA BLANCA / CORTOPUNZANTE                 A PIE         A PIE  35.0   
3819                ARMA DE FUEGO  PASAJERO MOTOCICLETA         A PIE  26.0   
3852  ARMA BLANCA / CORTOPUNZANTE                 A PIE         A PIE  35.0   
4785  ARMA BLANCA / CORTOPUNZANTE                 A PIE         A PIE  25.0   

           Sexo Estado civil País de nacimiento Clase de empleado Profesión  \
870   MASCULINO            -                  -        NO REPORTA         -   
982   MASCULINO            -                  -        NO REPORTA         -   
2213  MASCULINO            -                  -        NO REPORTA         -   
2630  MASCULINO            -                  -        NO REPORTA         -   
...         ...          ...                ...               ...       ...   
3447  MASCULINO            -           COLOMBIA       DESEMPLEADO         -   
3819  MASCULINO            -                  -        NO REPORTA         -   
3852  MASCULINO            -                  -        NO REPORTA         -   
4785  MASCULINO            -                  -        NO REPORTA         -   

     Escolaridad  Código DANE  Cantidad  
870            -    5837000.0         1  
982            -    5893000.0         1  
2213           -    5380000.0         1  
2630           -   54001000.0         1  
...          ...          ...       ...  
3447           -   68001000.0         1  
3819           -   81794000.0         1  
3852           -    5001000.0         1  
4785           -   25754000.0         1  

[13 rows x 20 columns]

By analysing 'Estado civil' for missing data, I consider there is no need for deleting these missing rows, these do not represent a critical feature at the momement.

In [29]:
homicidios_2018['País de nacimiento'].value_counts()

COLOMBIA     4828
VENEZUELA      91
-              35
ECUADOR         3
             ... 
NORUEGA         1
ESPAÑA          1
ITALIA          1
PERU            1
Name: País de nacimiento, Length: 10, dtype: int64

In [30]:
homicidios_2018['Clase de empleado'].value_counts()

EMPLEADO PARTICULAR       1654
INDEPENDIENTE             1515
DESEMPLEADO                847
AGRICULTOR                 331
                          ... 
EMPLEADO ARMADA/MARINA       2
DEPORTISTA                   2
POLITICO                     1
AFRODESCENDIENTE             1
Name: Clase de empleado, Length: 25, dtype: int64

'Profesion' is a critical variable given that more than 98% of data contains this piece of information and there is also no reported data whitin this group, this variable does not allow us to develop a proper data analysis

In [31]:
homicidios_2018['Profesión'].value_counts()

-                                               4837
NO REPORTADA                                      47
TECNICO PROFESIONAL EN SERVICIO DE POLICIA        24
POLICIAS                                          13
                                                ... 
ADMINISTRACION  POLICIAL                           1
PROFESORES DE EDUCACION MEDIA                      1
PUBLICIDAD                                         1
TECNICO PROFESIONAL EN INGENIERIA INDUSTRIAL       1
Name: Profesión, Length: 32, dtype: int64

In [32]:
(homicidios_2018['Profesión'] == '-').value_counts()

True     4837
False     127
Name: Profesión, dtype: int64

In [33]:
homicidios_2018['Escolaridad'].value_counts()

SECUNDARIA      2936
PRIMARIA        1709
ANALFABETA       107
TECNICO           73
SUPERIOR          68
NO REPORTADO      53
-                 14
TECNOLOGO          4
Name: Escolaridad, dtype: int64

In [34]:
homicidios_2018['Código DANE'].value_counts()

76001000.0    489
11001000.0    431
5001000.0     201
8001000.0     120
             ... 
5138000.0       1
68575000.0      1
68573000.0      1
23855000.0      1
Name: Código DANE, Length: 613, dtype: int64

In [35]:
(homicidios_2018['Código DANE'] == '-').value_counts()

C:\Programs\Anaconda3\lib\site-packages\pandas\core\ops.py:1649: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  result = method(y)


False    4964
Name: Código DANE, dtype: int64

'Cantidad' is a variable which is not quite clear given that the data reported has variables which describes only 1 person, so I would rather not considering this feature for our analysis

In [36]:
homicidios_2018['Cantidad'].value_counts()

1    4956
2       7
3       1
Name: Cantidad, dtype: int64

In [37]:
homicidios_2018[homicidios_2018['Cantidad'] == 3]

Fecha Departamento Municipio     Día      Hora  \
1569  10/04/2018 12:00:00 AM        CAUCA     PATÍA  Martes  18:00:00   

           Barrio   Zona Clase de sitio  Arma empleada Móvil Agresor  \
1569  VDA BETANIA  RURAL    LOTE BALDIO  ARMA DE FUEGO         A PIE   

     Móvil Victima  Edad       Sexo Estado civil País de nacimiento  \
1569         A PIE  35.0  MASCULINO      SOLTERO           COLOMBIA   

     Clase de empleado Profesión Escolaridad  Código DANE  Cantidad  
1569       DESEMPLEADO         -    PRIMARIA   19532000.0         3

In [38]:
homicidios_2018.drop('Cantidad', axis=1, inplace=True)

In [39]:
homicidios_2018.drop('Profesión', axis=1, inplace=True)

# Final dataset for data analysis

To have a better data handling

In [40]:
homicidios_2018.rename( columns = { 'Fecha' : 'FECHA', 'Departamento' : 'DEPARTAMENTO', 'Municipio' : 'MUNICIPIO',
                                   'Día' : 'DIA', 'Hora' : 'HORA', 'Clase de sitio' : 'CLASE SITIO', 'Barrio' : 'BARRIO',
                                   'Zona' : 'ZONA', 'Arma empleada' : 'ARMA EMPLEADA', 'Móvil Agresor' : 'MOVIL AGRESOR', 
                                   'Móvil Victima' : 'MOVIL VICTIMA', 'Edad' : 'EDAD', 'Sexo' : 'GENERO',
                                   'Estado civil' : 'ESTADO CIVIL', 'País de nacimiento' : 'PAIS NACIMIENTO',
                                   'Clase de empleado' : 'CLASE EMPLEADO', 'Escolaridad' : 'ESCOLARIDAD',
                                   'Código DANE' : 'CODIGO DANE'}, inplace = True )
homicidios_2018['DIA'].replace({ 'Lunes' : 'LUNES', 'Martes' : 'MARTES', 'Miércoles' : 'MIERCOLES', 'Jueves' : 'JUEVES',
                              'Viernes' : 'VIERNES', 'Sábado' : 'SABADO', 'Domingo' : 'DOMINGO'}, inplace = True)
homicidios_2018.replace({'-' : 'NO REPORTADO'}, inplace = True)

In [41]:
homicidios_2018

FECHA DEPARTAMENTO            MUNICIPIO        DIA  \
0     01/01/2018 12:00:00 AM     AMAZONAS         LETICIA (CT)      LUNES   
1     01/01/2018 12:00:00 AM     AMAZONAS         LETICIA (CT)      LUNES   
2     01/01/2018 12:00:00 AM    ANTIOQUIA                ANDES      LUNES   
3     01/01/2018 12:00:00 AM    ANTIOQUIA                ANDES      LUNES   
...                      ...          ...                  ...        ...   
4961  12/12/2018 12:00:00 AM    SANTANDER      BARRANCABERMEJA  MIERCOLES   
4962  12/12/2018 12:00:00 AM        VALLE            CALI (CT)  MIERCOLES   
4963  12/12/2018 12:00:00 AM        VALLE            CALI (CT)  MIERCOLES   
4964  12/12/2018 12:00:00 AM        VALLE  GUADALAJARA DE BUGA  MIERCOLES   

          HORA                   BARRIO    ZONA         CLASE SITIO  \
0      9:45:00                ESPERANZA  URBANA       VIAS PUBLICAS   
1      9:45:00                ESPERANZA  URBANA       VIAS PUBLICAS   
2      8:30:00          ALTO DEL CEDRON   RURAL       VIAS PUBLICAS   
3      2:30:00         PARQUE PRINCIPAL   RURAL             PARQUES   
...        ...                      ...     ...                 ...   
4961  20:09:00    MIRADORES DEL CACIQUE  URBANA       VIAS PUBLICAS   
4962  19:47:00     REPUBLICA ISRAEL E16  URBANA       VIAS PUBLICAS   
4963  19:47:00     REPUBLICA ISRAEL E16  URBANA       VIAS PUBLICAS   
4964  17:45:00  CORR. CHAMBIMBAL/CAMPIÑ   RURAL  FINCAS Y SIMILARES   

                    ARMA EMPLEADA         MOVIL AGRESOR MOVIL VICTIMA  EDAD  \
0     ARMA BLANCA / CORTOPUNZANTE                 A PIE         A PIE  16.0   
1     ARMA BLANCA / CORTOPUNZANTE                 A PIE         A PIE  17.0   
2                   ARMA DE FUEGO                 A PIE         A PIE  26.0   
3     ARMA BLANCA / CORTOPUNZANTE                 A PIE         A PIE  30.0   
...                           ...                   ...           ...   ...   
4961                ARMA DE FUEGO  PASAJERO MOTOCICLETA         A PIE  58.0   
4962                ARMA DE FUEGO                 A PIE         A PIE  32.0   
4963                ARMA DE FUEGO                 A PIE         A PIE  39.0   
4964  ARMA BLANCA / CORTOPUNZANTE                 A PIE         A PIE  16.0   

         GENERO ESTADO CIVIL PAIS NACIMIENTO       CLASE EMPLEADO ESCOLARIDAD  \
0     MASCULINO      SOLTERO          BRASIL  EMPLEADO PARTICULAR  SECUNDARIA   
1     MASCULINO      SOLTERO          BRASIL  EMPLEADO PARTICULAR  SECUNDARIA   
2     MASCULINO      SOLTERO        COLOMBIA        INDEPENDIENTE    PRIMARIA   
3     MASCULINO      SOLTERO        COLOMBIA           AGRICULTOR    PRIMARIA   
...         ...          ...             ...                  ...         ...   
4961  MASCULINO      SOLTERO        COLOMBIA  EMPLEADO PARTICULAR    PRIMARIA   
4962   FEMENINO      SOLTERO        COLOMBIA          AMA DE CASA  SECUNDARIA   
4963  MASCULINO      SOLTERO        COLOMBIA  EMPLEADO PARTICULAR  SECUNDARIA   
4964  MASCULINO      SOLTERO        COLOMBIA          DESEMPLEADO  SECUNDARIA   

      CODIGO DANE  
0      91001000.0  
1      91001000.0  
2       5034000.0  
3       5034000.0  
...           ...  
4961   68081000.0  
4962   76001000.0  
4963   76001000.0  
4964   76111000.0  

[4964 rows x 18 columns]

## In conlcusion for point 2, there has been a profound analysis of each of the elements of the dataset, this is very sensible data considering that it involves homicides data of Colombia, therefore I have found two very critical variables which were deleted from the final table, these are 'Profesion' and 'Cantidad'. I consider, it is important to have clear data and the source of it should be contacted if there are misleading information. 

## There is another aspect which is important to consider and it is the '-' or not given data, although I do not see it as a critical aspect to leave it out of the dataset, it should be considered if there is further analysis where those become an important part of it.

In [42]:
homicidios_2018.to_csv('./modified_dataset/homicidios_2018.csv', index=False)